src/components/pokemon/pokemon.component.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { APIcatchPokemon, APIsetPokemonFree } from 'FetchAPI/fetch.methods';
+import { catchPokemon } from 'Redux/pokemons/pokemons.actions';
+import { setPokemonFree } from 'Redux/pokemons/pokemons.actions';
+
+import CustomButton from 'Components/custom-button/custom-button.component';
+
+import './pokemon.styles.scss';
+
+class Pokemon extends React.Component {
+  render() {
+    const {
+      id,
+      name,
+      status,
+      captureDate,
+      catchPokemon,
+      setPokemonFree
+    } = this.props;
+    return (
+      <div
+        className={`${
+          status !== 'free' ? 'Pokemon_captured' : 'Pokemon_captured-alt'
+        } Pokemon`}
+      >
+        <div
+          className={`${
+            status !== 'free' ? 'hidden-background_alt' : 'hidden-background'
+          }`}
+        />
+        <Link to={`pokemon/${id}`}>
+          <img
+            className='Pokemon__image'
+            src={`../../assets/images/pokemons-images/${id}.png`}
+            alt='pokemon'
+          />
+        </Link>
+        <div className='Pokemon__text'>
+          <div className='Pokemon__id'>
+            ID: <span className='Pokemon__text_focus'>{id}</span>
+          </div>
+          <div className='Pokemon__name'>
+            Name:{' '}
+            <span className='Pokemon__text_focus'>
+              {name.split('')[0].toUpperCase() + name.slice(1)}
+            </span>
+          </div>
+          <div className='Pokemon__status'>
+            Status: <span className='Pokemon__text_focus'>{status}</span>
+          </div>
+          {status !== 'free' ? (
+            <div className='Pokemon__captureDate'>
+              Capture Date:{' '}
+              <span className='Pokemon__text_focus'>{captureDate}</span>
+            </div>
+          ) : null}
+        </div>
+        {status === 'free' ? (
+          <CustomButton
+            onClick={() => {
+              catchPokemon({ id, name, status, captureDate });

```

Comment:
Do you change internal state firstly, and after that just send request to BE? - it's totally incorrect - UI should be consistent with the model of application. The common flow in any app looks like - user make some action that should change model of app (in your case - click catch button) -> FE send request to the BE and wait for response -> if response is SUCCESS(200, 201 and so on) -> update state according BE response (in general, BE always return updated/created object).

Reply:
I see the problem. Currently I'm just changing internal state ignoring the possibility of request to BE be failed. Got it, should wait till my app will get the response and do stuff according to it's value.